packages/instrumentation-aws-sdk/src/services/bedrock-runtime.ts

```diff
@@ -579,4 +592,214 @@ export class BedrockRuntimeServiceExtension implements ServiceExtension {
       }
     }
   }
+
+  private async responseHookInvokeModelWithResponseStream(
+    response: NormalizedResponse,
+    span: Span,
+    tracer: Tracer,
+    config: AwsSdkInstrumentationConfig
+  ): Promise<any> {
+    const stream = response.data?.body;
+    const modelId = response.request.commandInput?.modelId;
+    if (!stream || !modelId) return;
+
+    // Replace the original response body with our instrumented stream.
+    // - Defers span.end() until the entire stream is consumed
+    // This ensures downstream consumers still receive the full stream correctly,
+    // while OpenTelemetry can record span attributes from streamed data.
+    response.data.body = async function* (
+      this: BedrockRuntimeServiceExtension
+    ) {
+      try {
+        for await (const chunk of stream) {
+          const parsedChunk = this.parseChunk(chunk?.chunk?.bytes);
+
+          if (!parsedChunk) return;
```

Comment:
@yuliia-fryshko 

nit: If, for whatever reason, a chunk cannot be parsed for adding otel data, should we perhaps *continue* to yield the chunks and eventually span.end() rather than `return;` here?

Reply:
I think your change in commit 80d93e3 is not correct. It would result in `span.end()` being called and then an attempt to call `BedrockRuntimeServiceExtension.record*Attributes(...)` which would likely break.

I've change it to what I think it should be in commit ee6f51e9.
@yuliia-fryshko Please let me know if my change looks inappropriate.